pre_commit_hooks/check_vcs_permalinks.py

```diff
@@ -25,11 +24,23 @@ def _check_filename(filename: str) -> int:
 def main(argv: Optional[Sequence[str]] = None) -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument('filenames', nargs='*')
+    parser.add_argument(
+        '--github-enterprise-domain', dest='github_enterprise_domain',
+    )
     args = parser.parse_args(argv)
 
     retv = 0
+
+    github_non_permalink = re.compile(NON_PERMALINK % b'github.com')
     for filename in args.filenames:
-        retv |= _check_filename(filename)
+        retv |= _check_filename(filename, github_non_permalink)
+
+    if args.github_enterprise_domain:
+        github_enterprise_non_permalink = re.compile(
+            NON_PERMALINK % args.github_enterprise_domain.encode(),
```

Comment:
please don't use %s, if anything use .format() or f-strings -- thanks!

I notice this is a bytestring which doesn't support .format() or f-strings, but you can change it to not be a bytestring and then do .encode() afterwards as you've done for the domain

you can probably also inline the string as well

Reply:
Since the `NON_PERMALINK` is used only inside the function that converts the domain to a regular expression, I changed it to inline and used the f-string.